File: PC^2/pc2/samps/practice.java
import java.io.*;
import java.text.*;

//
// File:    practice.java
// Purpose: practice program, sigma positive integers 
// Author:  pc2@ecs.csus.edu or http://www.ecs.csus.edu/pc2
// 
// Thu Jul 03 16:03:50 2003
// 
// $Id: practice.java,v 1.1 2005/03/06 05:34:39 laned Exp $
// 

public class practice {
    public static void main(String[] args) 
    {
	try
	{	
		RandomAccessFile file = new RandomAccessFile("practice.in", "r");
		String line;

		int inval = 0;
		int sum = 0;

		while((line = file.readLine()) != null) 
		{			
			inval = new Integer(line.trim()).intValue();
			sum = 0;

			if (inval == 0)
				System.exit (4);

			if (inval < 1)
			{
				for (int i = 1; i >= inval; i --)
				       sum += i;	
			}
			else
			{
				for (int i = 1; i <= inval; i ++)
				       sum += i;	
			}
			System.out.println("N = "+inval+"  Sum = "+sum);
		}

		System.out.println("Did not find trailing zero");
	}
	catch(Exception e)
	{
		System.out.println("Possible trouble reading practice.in");
	}			
    }
}

// eof practice.java $Id: practice.java,v 1.1 2005/03/06 05:34:39 laned Exp $

File: PC^2/pc2board/samps/practice.c

#include <stdio.h>
#include <stdlib.h>

/*
 * File:    practice.c
 * Purpose: sum of positive integers practice problem
 * Author:  pc2@ecs.csus.edu
 *
 * $Id: practice.c,v 1.1 2005/03/06 05:34:39 laned Exp $
 */

#define INFILE "practice.in"

int main (int argc, char **argv)
{
	int sum = 0;
	int i = 0;
	FILE *fp = fopen (INFILE,  "rt");
	char line[128];

	if (fp != NULL)
	{
		while ( fgets (line, sizeof(line), fp) != NULL)
		{
			long inval = atol (line);
			sum = 0;

			if (inval == 0)
				exit (4);

			if (inval < 1)
			{
				for (i = 1; i >= inval; i --)
				       sum += i;	
			}
			else
			{
				for (i = 1; i <= inval; i ++)
				       sum += i;	
			}
			printf("N = %-3d    Sum = %d\n", inval, sum);
		}

		printf("Did not find trailing zero \n");

	}
	else
	{
		fprintf(stderr,"Could not read file %s \n", INFILE);
	}

	exit (4);
}

/* eof practice.c */
